ext/bcmath/libbcmath/src/num2str.c

```diff
@@ -64,8 +63,15 @@ zend_string *bc_num2str_ex(bc_num num, size_t scale)
 	if (scale > 0) {
 		*sptr++ = '.';
 		sptr = bc_copy_bcd_val(sptr, nptr, nptr + min_scale);
-		for (index = num->n_scale; index < scale; index++) {
-			*sptr++ = BCD_CHAR(0);
+
+		size_t scale_diff;
+		if (scale > num->n_scale && (scale_diff = scale - num->n_scale) > 8) {
+			memset(sptr, BCD_CHAR(0), scale_diff);
+			sptr += scale_diff;
+		} else {
+			for (size_t index = num->n_scale; index < scale; index++) {
+				*sptr++ = BCD_CHAR(0);
+			}
```

Comment:
It's safe to always compute scale_diff, the compiler will do the optimization for you, and it will also be cleaner code.
Anyway, this part didn't make a real difference on my system, the UNEXPECTED part did however make an improvement.

To have more stable measurements, I recommend using [hyperfine](https://github.com/sharkdp/hyperfine) which is what I typically use. It runs the code multiple times and also computes a confidence interval. It can also do warmup rounds to reduce noise. I found the results from this tool to be more stable than only using the benchmark script. To use it here I would recommend to reduce the number of iterations in your benchmark script (to make sure the total runtime is below a second to avoid CPU throttling), and to remove the timing from the benchmark script and only rely on hyperfine instead.